core/src/main/java/com/netflix/conductor/core/execution/WorkflowExecutor.java

```diff
@@ -1620,7 +1619,7 @@ public void skipTaskFromWorkflow(
             taskToBeSkipped.setOutputMessage(skipTaskRequest.getTaskOutputMessage());
         }
         executionDAOFacade.createTasks(Collections.singletonList(taskToBeSkipped));
-        decide(workflow);
+        decide(workflow.getWorkflowId());
```

Comment:
What if we wrap invoking `decide(WorkflowModel)` within a lock here? Since we already have the workflow loaded in memory here, we dont need to perform this again.

Reply:
`decide(workflowId)` calls `decide(WorkflowModel)` and the former is already has a lock. `decide(WorkflowModel)` is also called recursively which means the lock needs to be re-entrant. Since `skipTaskFromWorkflow` is used sparingly, the additional call to load `WorkflowModel` shouldn't have a big impact.